index.ts

```diff
@@ -68,14 +69,30 @@ export class HttpClientResponse implements ifm.IHttpClientResponse {
     public message: http.IncomingMessage;
     readBody(): Promise<string> {
         return new Promise<string>(async (resolve, reject) => {
-            let output = Buffer.alloc(0);
-
-            this.message.on('data', (chunk: Buffer) => {
-                output = Buffer.concat([output, chunk]);
-            });
+            let buffer: Buffer = Buffer.alloc(0);
+            const encodingCharset = utils.obtainContentCharset(this);
+
+            // Extract Encoding from header: 'content-encoding' and check against the supported BufferEncoding types
+            // Match `gzip`, `gzip, deflate` variations of GZIP encoding
+            const contentEncodingHeader: string = this.message.headers['content-encoding'] || '';
+            const isGzippedEncoded: boolean = new RegExp('(gzip$)|(gzip, *deflate)').test(contentEncodingHeader);
+
+            const supportedEncodings = ['ascii', 'utf8', 'utf-8', 'utf16le', 'ucs2', 'ucs-2', 'base64', 'latin1', 'binary', 'hex']
```

Comment:
If you look at the original PR in `microsoft/typed-rest-client` you won't see these changes, this is largely because we're using a newer version of TS and some newer type definitions. Previously you could do `Buffer.from(string,string)` but now you have to do `Buffer.from(string, BufferEncoding)`

`BufferEncoding` is a type so you can't directly infer if matches one of the valid subsets of `BufferEncoding` so I'm having to use `includes`. This thread explains the issue really really well: https://stackoverflow.com/questions/51528780/typescript-check-typeof-against-custom-type